tests/integration/test.backoff.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+var adapters = [
+    ['local', 'http']
+];
+
+adapters.forEach(function (adapters) {
+    var suiteName = 'test.backoff.js-' + adapters[0] + '-' + adapters[1];
+    describe(suiteName, function () {
+
+        var dbs = {};
+
+        beforeEach(function (done) {
+            dbs.name = testUtils.adapterUrl(adapters[0], 'testdb');
+            dbs.remote = testUtils.adapterUrl(adapters[1], 'test_repl_remote');
+            testUtils.cleanup([dbs.name, dbs.remote], done);
+        });
+
+        after(function (done) {
+            testUtils.cleanup([dbs.name, dbs.remote], done);
+        });
+
+        it('backoff event listeners', function (done) {
+            var remote = new PouchDB(dbs.remote);
+            var db = new PouchDB(dbs.name);
+            const rep = db.sync(remote, {
+                live: true,
+                retry: true,
+                heartbeat: 1,
+                timeout: 1,
+                back_off_function: function () { return 1; }
+            });
+
+            setTimeout(function () {
+              var numAListeners = rep.pull.listeners("active").length;
+              numAListeners.should.be.below(3);
+              done();
+            }, 600);
+        });
```

Comment:
Hey the feedback is great!  I'll try and get all this stuff working.  I couldn't get the 'shoulds'
to throw the assertion in the listeners. It just halted the flow.  Any suggestions to avoid the setTimeout are appreciated. 
Thank youJames
------
@cgul
atVenu.com




On Tue, Jun 28, 2016 at 3:40 AM -0600, "Dale Harvey" <notifications@github.com> wrote:












In tests/integration/test.backoff.js:
> +            var remote = new PouchDB(dbs.remote);
> +            var db = new PouchDB(dbs.name);
> +            const rep = db.sync(remote, {
> +                live: true,
> +                retry: true,
> +                heartbeat: 1,
> +                timeout: 1,
> +                back_off_function: function () { return 1; }
> +            });
> +
> +            setTimeout(function () {
> +              var numAListeners = rep.pull.listeners("active").length;
> +              numAListeners.should.be.below(3);
> +              done();
> +            }, 600);
> +        });



So this looks like a good start however there is a main problem in that the replication will still be active when the test finishes, we havent been using es6 features like const around so the test will need to look something like

    var rep = db.sync(remote, {
      live: true,
      retry: true,
      heartbeat: 1,
      timeout: 1,
      back_off_function: function () { return 1; }
    });

    rep.on('complete', function() { done(); });

    setTimeout(function () {
      var numAListeners = rep.pull.listeners("active").length;
      numAListeners.should.be.below(3);
      rep.cancel();
    }, 600);




If we could get rid of the setTimeout that would be perfect (count the backoffs?) however if that doesnt work it is not the wost thing in the world. Last tiny nit is that we use 2 space tabs, not 4



—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.